src/jit/lsra.cpp

```diff
@@ -8581,20 +8570,16 @@ void LinearScan::insertMove(
     // This var can't be marked lvRegister now
     varDsc->lvRegNum = REG_STK;
 
-    var_types lclTyp = varDsc->TypeGet();
-    if (varDsc->lvNormalizeOnStore())
-    {
-        lclTyp = genActualType(lclTyp);
-    }
+    var_types lclTyp = genActualType(varDsc->TypeGet());
```

Comment:
> Yes, and case 2 is the case that is problematic because we will need a byteable register for a byte type.

Why would we need a byte-addressable register in this case? Won't we just generate a `mov{z,s}x r32, m8`?